src/quick-lint-js/fe/parse-statement.cpp

```diff
@@ -2824,6 +2843,13 @@ void Parser::parse_and_visit_switch(Parse_Visitor_Base &v) {
       break;
 
     case Token_Type::kw_case: {
+      if (!is_before_first_switch_case &&
+          !is_valid_end_of_case(previous_statement_first_token.type) &&
```

Comment:
[GCC reports a warning](https://github.com/quick-lint/quick-lint-js/actions/runs/6820808348/job/18550287972?pr=1094) because it thinks `previous_statement_first_token` might be uninitialized here.

To fix the warning, I think making `previous_statement_first_token` a `std::optional` is a decent solution.